### ISSUE TITLE
[FIRRTL] Empty port names can't be roundtripped

### ISSUE THREAD

**Issue by darthscsi** [llvm/circt#1822]
modules made with empty names for arguments print out like this, which then doesn't parse.
```
hw.module @foo(%arg1: i32) attributes {argNames = [""]} {
}
```

**Comment by darthscsi** [comment]
Similarly with duplicate port names.  Anything that causes argNames to be printed.

**Comment by lattner** [comment]
My bug, I'll fix

**Comment by lattner** [comment]
Fixed in 13deb2be, thanks!
